Sources/Private/CoreAnimation/Animations/CALayer+addAnimation.swift

```diff
@@ -320,3 +407,52 @@ extension CALayer {
   }
 
 }
+
+// MARK: - KeyframeProtocol
+
+/// We can't do `extension Array where Element == Keyframe` since we don't have a way
+/// to specify the generic type of the keyframe itself. That would require a syntax
+/// like `extension <T> Array where Element == Keyframe<T>`. Instead we can use
```

Comment:
remember in like Swift 3 or earlier when you had to do this any time you wanted to extend an array with a specific `Element` (since there wasn't an `Element == Foo` syntax yet)

Reply:
I assume this doesn't work as the following? Or are you trying to avoid a function and have this be a computed property that doesn't support where clauses:
```swift
extension Array {
  func splitByCalculationModeSegment<Value>() 
    -> [[Keyframe<Value>]] 
    where
    Element == Keyframe<Value>
  {
    …
  }
}
```

Reply:
ooooohhh good idea, let me try that

Reply:
much better, thanks for the clever suggestion! 3f14b41